yt_dlp/extractor/instagram.py

```diff
@@ -367,47 +367,57 @@ def _real_extract(self, url):
         video_id, url = self._match_valid_url(url).group('id', 'url')
         media, webpage = {}, ''
 
+        if self._get_cookies(url).get('sessionid'):
+            info = traverse_obj(self._download_json(
+                f'{self._API_BASE_URL}/media/{_id_to_pk(video_id)}/info/', video_id,
+                fatal=False, errnote=False, note='Downloading video info', headers={
+                    **self._API_HEADERS
+                }), ('items', 0))
+            if info:
+                media.update(info)
+                comments_info = self._download_json(
+                    f'{self._API_BASE_URL}/media/{_id_to_pk(video_id)}/comments/?can_support_threading=true&permalink_enabled=false', video_id,
+                    fatal=False, errnote=False, note='Downloading comments info', headers={
+                        **self._API_HEADERS
+                    }) or {}
```

Comment:
```suggestion
                    f'{self._API_BASE_URL}/media/{_id_to_pk(video_id)}/comments/?can_support_threading=true&permalink_enabled=false',
                    video_id, fatal=False, errnote=False, note='Downloading comments info',
                    headers=self._API_HEADERS) or {}
```
1. Why errnote=False?
2. If the only point of this API call is to get comments, it should go inside `_get_comments` and be called post-extraction

Reply:
so you are saying that comments extraction must be moved to a function called `_get_comments`? and what are you saying about `post-extraction`? I didn't get it

Reply:
yt-dlp has an internal framework to prevent additional requests for comments/subtitles unless user requests it. To make use of this, define a function `_get_comments(...)` that makes all the comment API calls and `yield {comment}`s. Then in the infodict, put `'__post_extractor': self.extract_comments(...)`

Reply:
See any of these for example implementation
![image](https://user-images.githubusercontent.com/6623388/187969294-03ff7c42-a2d0-474f-bb52-87c1a68ab4ae.png)